tracy-client/src/lib.rs

```diff
@@ -19,6 +19,27 @@ use std::alloc;
 use std::ffi::CString;
 use tracy_client_sys as sys;
 
+/// Start a new Tracy span with function, file, and line determined automatically.
+#[macro_export]
+macro_rules! span {
+    ($name:expr, $callstack_depth:expr) => {{
+        fn f() {}
+        fn type_name_of<T>(_: T) -> &'static str {
+            std::any::type_name::<T>()
+        }
```

Comment:
From what I can tell this can be simplified to a

```rust
struct S; 
let func_name = std::any::type_name::<S>();
let function = &func_name[..func_name.len() - 3];
```